atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -437,11 +440,39 @@ private void abortPendingSnapshots() {
       return future;
     }
 
+    if (!checkChecksums(request, future)) {
+      return future;
+    }
+
     // Append the entries to the log.
     appendEntries(request, future);
     return future;
   }
 
+  private boolean checkChecksums(
+      final AppendRequest request, final CompletableFuture<AppendResponse> future) {
+    if (request.checksums() != null) {
+      boolean failed = false;
+      if (request.entries().size() != request.checksums().size()) {
+        log.debug("Rejected {}: expected the same number of checksums as entries", request);
```

Comment:
Also add request size and checksum size to the log message.

Reply:
That's what the request prints 
```
@Override
  public String toString() {
    return toStringHelper(this)
        .add("term", term)
        .add("leader", leader)
        .add("prevLogIndex", prevLogIndex)
        .add("prevLogTerm", prevLogTerm)
        .add("entries", entries.size())
        .add("checksums", checksums.size())
        .add("commitIndex", commitIndex)
        .toString();
  }
```